src/Workspaces/Core/Portable/Workspace/Solution/AssetPath.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.Serialization;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Serialization;
+
+/// <summary>
+/// Required information passed with an asset synchronization request to tell the host where to scope the request to. In
+/// particular, this is often used to scope to a particular <see cref="Project"/> or <see cref="Document"/> to avoid
+/// having to search the entire solution.
+/// </summary>
+[DataContract]
+internal readonly struct AssetPath
+{
+    /// <summary>
+    /// Instance that will only look up solution-level data when searching for checksums.
+    /// </summary>
+    public static readonly AssetPath SolutionOnly = AssetPathKind.Solution;
```

Comment:
>SolutionOnly [](http://example.com/codeflow?start=37&length=12)

nit: doesn't seem necessary. Almost all other callers just specify the AssetPathKind explicitly.